manifests/user.pp

```diff
@@ -6,6 +6,7 @@
   $primary_group = "${title}", # intentionally, workaround for: https://tickets.puppetlabs.com/browse/PUP-4332
   $comment = "${title}", # see https://github.com/deric/puppet-accounts/pull/11
   $username = "${title}",# for more details
+  $group_members = undef,
```

Comment:
This is quite confusing. Do we really want to add members into user's primary group? And how do we synchronize the data?
```puppet
  class {'accounts':
    groups => {
      'users' => {
        'members' => ['john'],
      },
    },
    users => { 'john' => {
      'groups'  => ['users'],
      'group_members' = ['snow']
    }}
  }
```
In this case user John should be a member of group `users`. That's clear, even specified twice (once in user's definition and other in group definition). Then we have undefined user `snow`, which should be created? or it should throw an error? Do I have to specify user itself in his primary group?

Could you write some tests for this functionality? What if we want to remove users from some group?

We have multiple attributes concerning groups, `primary_group`, `gid`, `groups` and now `group_members`. It's starting to be quite confusing :(